### PR TITLE
Fix clippy::large_enum_variant and add to DENIED_LINTS

### PR DESCRIPTION
**Pull request by baumanj**
See https://rust-lang.github.io/rust-clippy/master/#large_enum_variant

Some related fixes as well, view by commit for full explanation.

Clippy tracking issue: https://github.com/habitat-sh/habitat/issues/6213

### CODE REVIEW

File: components/butterfly/build.rs [hunk 1]
```diff
@@ -14,6 +14,8 @@ fn main() {
 
 fn generate_protocols() {
     let mut config = prost_build::Config::new();
+    config.type_attribute(".butterfly.newscast.Rumor.payload",
+                          "#[allow(clippy::large_enum_variant)]");
```
Comment (christophermaier) [inline components/butterfly/build.rs]: Nice one.

File: components/sup/src/cli.rs [hunk 2]
```diff
@@ -29,6 +28,7 @@ mod test {
 
     mod sup_run {
         use super::*;
+        use std::iter::FromIterator as _;
```
Comment (christophermaier) [inline components/sup/src/cli.rs]: That's an interesting new trick!

**Comment by baumanj** [comment]
The CI is going to fail until we get the build kite runner upgraded to rust 1.33.0

**Comment by baumanj** [comment]
Rebased now that both linux and windows CI images are on rust 1.33.0

**Pull request merged**